applications/adv_networking_bench/cpp/CMakeLists.txt

```diff
@@ -107,3 +101,19 @@ install(
         ../adv_networking_bench_rmax_rx_yaml
   DESTINATION bin/adv_networking_bench/cpp
 )
+
+# Add testing
+if(BUILD_TESTING)
+  file(READ "${CMAKE_CURRENT_SOURCE_DIR}/../adv_networking_bench_default_tx_rx.yaml" TEST_CONFIG)
+  string(REGEX REPLACE "eth_dst_addr:[^\n]*" "eth_dst_addr: notused" TEST_CONFIG ${TEST_CONFIG})
+  string(REGEX REPLACE "ip_src_addr:[^\n]*" "ip_src_addr: notused" TEST_CONFIG ${TEST_CONFIG})
+  string(REGEX REPLACE "ip_dst_addr[^\n]*" "ip_dst_addr: notused" TEST_CONFIG ${TEST_CONFIG})
+  string(REGEX REPLACE "max_duration_ms:[^\n]*" "max_duration_ms: 20000" TEST_CONFIG ${TEST_CONFIG})
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/adv_networking_bench_default_tx_rx_test.yaml" ${TEST_CONFIG})
+
+  # Add test
+  file(COPY "testing/adv_networking_test_harness.py" DESTINATION "testing")
+  add_test(NAME adv_networking_bench_dpdk_test
```

Comment:
I'm wondering about the choice to combine run + validate into one `add_test` entry. We have a few other instances where application runtime is handled in one test entry and validation (frames, logs, etc) are handled in a followup test entry. See [Endoscopy Tool Tracking](https://github.com/nvidia-holoscan/holohub/blob/main/applications/endoscopy_tool_tracking/cpp/CMakeLists.txt#L75) as one example where frames validation follows runtime.

Encapsulating both steps in one `add_test` with your custom test harness seems OK, thought it might add complexity with spawning the app subprocess and log redirection. Curious about your thoughts.

Reply:
Thanks for discussion offline. Current approach with test harness seems good and will be extensible with additional ANO-specific test validation steps, i.e. NIC throughput observation in parallel. Reasonable to deviate from other HoloHub CTest approaches for this case.